### DISCUSSION TITLE
Thoughts on gamification

### DISCUSSION THREAD

**Discussion by DoggoOfSpeed** [streetcomplete/StreetComplete#2598] [Ideas]
So I guess I have to explain why gamification would be very useful...

- It makes the app way more accessible. Telling your friends that this app is a way of improving OSM is way less effective than saying that it is just a game. As the invitees don't know what OSM is, they don't have a reason to help. Even if you explain how great and useful OSM is, they won't be nearly as interested as they would be if you just told them it is a game. (I actually had this happen to me :/)
- It makes contributing fun. Right now, you have to find the fun in answering questions yourself. Answering them isn't inherently fun, but even adding some sliders that indicate your progress would be great.
- It makes it more competitive. Right now you just know in which place you are locally and globally. Even a simple thing like an actual leader board would almost certainly encourage contributing. You are way more likely to contribute after seeing that you are just 3 stars away from advancing a place.

OK, so I hope you understand why gamification is cool, (If you disagree, do tell :)

One simple thing would be the leader board I mentioned, as I said it would encourage contributing.
Another thing I think would be good is showing your progress in sliders. Right now there are only achievements they are a good start, but the problem is that they just award people for contributing, not encourage it. A slider on the other hand encourages. As there are a lot of different types of achievements, there could be a lot of bars. While overwhelming the player might sound like a bad idea, the more sliders there are, the more are almost full. When the bar is almost full, it encourages the player to collect just a few stars, then they realize they almost filled another bar and the cycle continues.

All other ideas I had need friends and competitors, while this might sound iffy at best, you have to realize that those who play alone, most likely are already familiar with OSM, so they need less reasons to contribute, so they will most likely look at Street Complete as just another tool to do just that. The friends they invite on the other hand, need more reasons to contributing (as I said, explaining what OSM is most likely wont suffice), so they need StreetComplete to be fun.

So one Idea I had would be marking the places the players have contributed as kingdoms. The kingdom could be an outline of your area of contribution. If, for example, you have contributed a lot in one place, that area would become small, but strong, if you went around just marking the building's type, you would expand your kingdom, but it wouldn't be as strong. You could see people around you: your friends, even people you don't know. This would also act as encouragement. If you see a strong kingdom in your city, you will try to match it. If you see a small kingdom expand quickly, you will also be encouraged to keep your throne by expanding your walls.

You could talk with other kingdoms, maybe even make some friends in your city. Granted, I haven't thought about this idea for that long, so maybe it is bad and I just haven't realized it, but right now I think it is OK. 

I don't expect any of my ideas to actually be realized, but I just wanted to know what you thing about adding more game like elements to StreetComplete. I think I suck at expressing ideas, so If you disagree, agree or want to ask something, please do :)

**Comment by matkoniecz** [comment]
> So I guess I have to explain why gamification would be very useful...

Note that it is present already.

> One simple thing would be the leader board I mentioned, as I said it would encourage contributing.

See comments in https://github.com/streetcomplete/StreetComplete/issues/2541

**Reply by DoggoOfSpeed** [reply]
Fair enough... I never thought about that :/

**Comment by smichel17** [comment]
> It makes the app way more accessible. Telling your friends that this app is a way of improving OSM is way less effective than saying that it is just a game. As the invitees don't know what OSM is, they don't have a reason to help. Even if you explain how great and useful OSM is, they won't be nearly as interested as they would be if you just told them it is a game. (I actually had this happen to me :/)

This is a double-edged sword. A friend who is just looking at SC as a game may not care all that much for the accuracy of their answers, not realizing that their incorrect answers are polluting OSM data. So, at least in its current form, I think it is very important that mappers should know what the data is used for. I do agree that further gamification could help increase accessibility, you'd just have to make sure to deal with data quality issues.

For example, perhaps there could be a separate "StreetComplete game" which includes much more gamification. In this version, changes would not be uploaded directly to OSM; instead, they would be uploaded to a server dedicated to the SC game, and would only be sent to OSM when 2-3 people have independently verified the answer. This would be more work to maintain, however.

**Reply by westnordost** [reply]
That was the concept of [Kort](https://wiki.openstreetmap.org/wiki/Kort_Game). It failed because a critical mass for that was misssing. I.e. seldomly, there were more than one user solving quests in the same area.

**Reply by JLZIMMERMANN** [reply]
The problem of Kort game was that it was a serious game and not really a game.

**Comment by matkoniecz** [comment]
> would only be sent to OSM when 2-3 people have independently verified the answer

That was tried, and failed (among other reasons) due to not enough density of mappers. How many quests would be independently solved by multiple people within short time?

(I do not remember app name right now)

**Reply by HolgerJeromin** [reply]
Do you mean https://wiki.openstreetmap.org/wiki/Kort_Game ?

**Reply by matkoniecz** [reply]
Yes.

**Reply by smichel17** [reply]
I'm having trouble finding the answer without doing a source dive, so maybe someone quickly remembers: Did Kort have confirmation gamified? ie, make it so you get double or triple points for confirming an answer rather than being the first to answer?

From the screenshot it also looks like it would show all quests at once, whereas for a game that's not what you want. Also, it looks like StreetComplete is *much* more visually polished than Kort was. That may not matter as much for a mapping tool, but for a game it makes a difference.

So basically, it's interesting that it's been tried before, but I don't think it's proof that the concept can't work. But maybe it is proof that you can't just slap on a little gamification to make it work; there needs to be more holistic design work.

(Aside, @matkoniecz was your original comment here from email?)

**Reply by matkoniecz** [reply]
> (Aside, @matkoniecz was your original comment here from email?)

no

**Reply by tiptoptom** [reply]
One way to increase the probability of verification would be to limit the game to certain areas. That way you bundle the manpower.

**Comment by e-t-l** [comment]
Leaderboard, progress bars, and kingdoms are all creative ideas, but gamification could start out even simpler than that.

From a psychology perspective, our monkey-brains love shiny rewards. Even just some small sensory reward for each quest completed—a little shower of cartoon stars and a triumphant-sounding chime (all togglable options, of course) for example, perhaps with a bigger star/heart explosion and sound effect after every 10th quest—would give users just enough of a dopamine hit to make it feel really satisfactory to complete quests. (There's a reason Candy Crush is so damn popular despite being utterly devoid of substance).

The above would be 1) very straightforward to implement, 2) not require any additional servers, and 3) not run as high a risk of over-gamifying it the way that competitive features like leaderboards might.

**Reply by mnalis** [reply]
> dopamine hit to make it **feel really satisfactory** to complete quests. (There's a reason Candy Crush is so damn popular despite being utterly devoid of substance).

That is the problem, IMHO.  When some action becomes satisfactory in itself, there is a risk that correctness of the data falls to the bottom of priority list (or worse). So, if it makes SC like Candy Crush, it is probably not a good idea.

(ref: [SC FAQ: I like gamification factor of StreetComplete, and have an idea how to increase it even more](https://wiki.openstreetmap.org/wiki/StreetComplete/FAQ#I_like_gamification_factor_of_StreetComplete,_and_have_an_idea_how_to_increase_it_even_more))

As for sound effects, see [SC FAQ: How can I disable/enable sound effects?](https://wiki.openstreetmap.org/wiki/StreetComplete/FAQ#How_can_I_disable/enable_sound_effects?)

**Reply by westnordost** [reply]
Some more animations sound good, but if they are too in your face they will become somewhat distracting for users who want to solve many quests. Note that there is already a "quest solved" animation, and also a sound, though.
